core/plugin.go

```diff
@@ -85,6 +85,16 @@ var (
 		metricPluginsStopping,
 		metricPluginsDead,
 	}
+
+	// Has to be index parallel to PluginState*
+	stateToDescription = []string{
```

Comment:
I don't see how struct tags could work here; we need a mapping from enum values to strings, not string values for a struct's fields. It's analogous to the stateToMetric mapping above for the same enums.